lib/targets/cuda/device.cpp

```diff
@@ -115,6 +115,26 @@ namespace quda
       return device_count;
     }
 
+    void get_visible_devices_string(char device_list_string[128])
+    {
+      char *device_order_env = getenv("CUDA_VISIBLE_DEVICES");
+
+      if (device_order_env) {
+        std::stringstream device_list_raw(device_order_env); // raw input
+        std::stringstream device_list;                       // formatted (no commas)
+
+        int device;
+        while (device_list_raw >> device) {
+          // check this is a valid policy choice
+          if (device < 0) { errorQuda("Invalid CUDA_VISIBLE_DEVICE ordinal %d", device); }
```

Comment:
Should be CUDA_VISIBLE_DEVICES (missing s)

Reply:
It was missing in the old code :)

Reply:
0bfa855e260c932d787d2d005fceb7c8c4088176

Reply:
🤣